data/fintype.lean

```diff
@@ -586,4 +586,69 @@ lemma fintype.card_equiv [fintype α] [fintype β] (e : α ≃ β) :
   fintype.card (α ≃ β) = (fintype.card α).fact :=
 fintype.card_congr (equiv_congr (equiv.refl α) e) ▸ fintype.card_perm
 
-end equiv
\ No newline at end of file
+end equiv
+
+section choose_unique
+open fintype
+open equiv
+
+variables [fintype α] [decidable_eq α] (p : α → Prop) [decidable_pred p]
+
+def choose_unique_x (p_eu : ∃! a : α, p a) : {a // p a} :=
```

Comment:
You shouldn't use any tactics in a definition unless you know what you're doing, I think.